benchmark/README.md

```diff
@@ -0,0 +1,6 @@
+Benchmarks here are generally in src/main and can be run with `benchmark/jmh:run`
+within SBT.
+
+See here for more details:
```

Comment:
This isn't a PR to actually merge as is imo. Its really guidance for "when someone goes to make a benchmark", please use this as a base.